client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -1921,7 +1914,7 @@ impl NetworkBehaviour for Notifications {
 							matches!(s, ConnectionState::Opening | ConnectionState::Open(_))
 						}) {
 							trace!(target: "sub-libp2p", "PSM <= Dropped({:?})", peer_id);
-							self.peerset.dropped(set_id, peer_id, DropReason::Refused);
+							self.protocol_controller_handles[usize::from(set_id)].dropped(peer_id);
```

Comment:
can protocol_controller_handles[set_id] go out of bounds?

Reply:
No, the list of notification protocols is never modified after initialization. And `set_id` is just an index in this list. I'm also not very happy that we refer to protocols by their index in the list, so the original idea was to keep a map of `ProtocolName` -> `ProtocolHandle`, but using a vector is more consistent with other places where we refer to notification protocols by index.